### ISSUE TITLE
Autosubmit GUI new release preparation

### ISSUE THREAD

**Issue by LuiggiTenorioK** [BSC-ES/autosubmit-gui#44]
In GitLab by @ jberlin on Oct 13, 2022, 13:02

Working path for the deployment of the new release of the Autosubmit GUI:

**Pre-Staging**

- [x] Update documentation for GUI and API in the wiki or readthedocs.io (if needed).
- [x] Update the news file located in src/components/pages/News.js with the latest changes done/ fixed bugs

**Deployment VM for Validation**

- [x] Generate the API new version and upload to Pypi.
- [x] Update GUI and deploy API in development environment for the final validation
- [x] Final validation of the release, Close all issues in the milestone, both in the GUI and API projects.

**Deployment in Production**

- [x] if validation in VM was OK, deploy in production, before that send email to distributions lists to warn the users of the update/possible disruptions
- [x] Validation in production, checking of the solved the issues & new features.

**Announce**

- [x] Prepare a message to send to the BSC-ES mailing list, with a brief summary of the new features (and also reminding abut the best advantages of using the GUI)

@mcastril @ggcr @manuel-g-castro

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Oct 17, 2022, 08:37

Regarding the new changelog that will go in the news section, I've drafted this from different changes. 

There are things that should be added such as @ jberlin new wrapper ordering, the favicon, a mention to the new dev env, etc...

Feel free to modify and delete any of the parts that you feel like don't belong here, or make any suggestions!

> **xx-10-2022**
> 
> This new release has been dedicated to improving some capabilities towards a more performant and better experience for our users. The main goal is to make the website more reactive and dynamic.
> 
> - In the Dashboard, we increased the retrieval speed for the experiments' Detailed Data. Now only experiments on the current page will be refreshed.
> - A new endpoint for stopping current background API jobs has been implemented (on the API side). Alongside this, we have a clean-up hook to use in all the required situations. 
> - The new endpoint allows discarding expensive, non-relevant petitions and dispatching new user demands on time. Thus, proving a better user experience.
> - Graph, Tree and QuickView start loading automatically, and the transition between them is improved due to the mentioned task canceling mechanism.
> - An early initial load of the Active experiments at the homepage has been implemented without requiring the user to click on the button each time.
> - Tasks without dependencies are displayed correctly in the Graph View in a single row.
> - Added redirection support after logging in.
> - Fixed a bug in the tree generation that was making the load more tedious.
> - Fixed a visual bug regarding the close button in the pop-up windows.
> - Mouse square selection for multiple graph nodes on Right Click. (Pending)

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Oct 17, 2022, 09:12

Also, is this repo Autosubmit GUI private? 

We should link the user here for issues rather than autosubmit repo.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Oct 17, 2022, 09:18

Yes, this repo is private. That is why I had to create another repo [just with the documentation](https://earth.bsc.es/gitlab/es/autosubmit-gui-docs/) at [add-documentation branch](https://earth.bsc.es/gitlab/es/autosubmitreact/-/tree/add-documentation). Maybe we can use [that other repo](https://earth.bsc.es/gitlab/es/autosubmit-gui-docs/) as issue tracker while we can't make this repo public.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 17, 2022, 09:22

Hi ! which is the repo that is private @ggcr ? am I wrong or it seems the link is missed ?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 17, 2022, 09:25

thanks @ggcr for preparing  the news part. I think for this week we can start with the preparations for the new release. from my side I need to re-check the label thing that we decided to add in addition to the fix of the ordering in the treeview

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 17, 2022, 13:23

Sorry I am lost. It's private but open to any Earth Sciences user. As the GUI is only used internally, what else would we need?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Oct 17, 2022, 13:31

![image](https://earth.bsc.es/gitlab/es/autosubmit-gui/uploads/268eb771b195cdb6783e474f2da7be48/image.png)

In the `About` page, at the last section at the bottom, the link named "Autosubmit GUI Issue page" redirects to the **Autosubmit** issues page. Shouldn't it redirect to **Autosubmit GUI** issues page?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Oct 17, 2022, 13:32

Yes, this is the only solution if this repo stays in private mode.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 17, 2022, 13:32

Thanks @ggcr. I did minor modifications on the text

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 17, 2022, 13:36

Yes, please change it.

This is legacy. Before, all issues were reported in Autosubmit.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 18, 2022, 17:57

Can you start with the Pre-staging @ jberlin ? Nothing prevents us from moving it right?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 18, 2022, 18:16

Hi @mcastril, I need to validate my latest changes for the treview and then I think we are OK for the pre-staging. I copied the a550 for this.we will need to clean up the development environment and install the release over there to test the involved issues without making any change.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 19, 2022, 10:13

> I need to validate my latest changes for the review

True, the labels in startdate level were missing. Please report the progress in the corresponding issue so that we can comment it tomorrow.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 24, 2022, 18:14

mentioned in commit ad221660bec80c6f0dc76b8260f7fae3dae51375

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 24, 2022, 18:14

marked the checklist item **Update the news file located in src/components/pages/News.js with the latest changes done/ fixed bugs** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 24, 2022, 18:15

I updated the News.JS tomorrow I will check the UI readthedoc.io to see if we are missing something

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 25, 2022, 10:39

Hi @manuel-g-castro , I was checking the readthedocs.io corresponding to the GUI, is it possible that some imgs are missed or the links broken ? 

https://autosubmit-gui.readthedocs.io/en/latest/experiment/index.html

I think we can add in the section of the tree, the improvements done concerning the UI/UX such as the ordering of the wrappers, the new folders and functionalities to expand/collapse all items.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 25, 2022, 12:22

marked the checklist item **Update documentation for GUI and API in the wiki or readthedocs.io (if needed).** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Oct 26, 2022, 08:40

Yeah, images were not uploaded to the repository because that is bad practice. I was supposed to find another way of making readthedocs fetch the images without having to upload them to the repo. Any ideas, @kinow?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @kinow on Oct 26, 2022, 09:29

If the images are static, I think they can go into the Sphinx [`html_static_path`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path) location.

But if the images are created dynamically, and depend on the project code, then +1 to not including the images directly. We can instead use Python to generate the images during the build (in readthedocs, their build system would generate the images for us).

There are a few different ways of achieving that, depending on the image type (png/jpg generated by the tool, a dot/graphviz graph, etc.). If this is blocking a release, maybe we can add the images in static for now and open a follow-up issue to generate the images dynamically.

Cheers

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Oct 26, 2022, 09:34

So we upload the images onto another service, like i dunno, drive, flicker, etc and put the folder path with images directly to the `html_static_path`? This could be very much useful for Autosubmit's Docs aswell.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Oct 26, 2022, 09:49

If you use static images you could explore uploading them to GitLab (by creating a wiki and upload them, they are stored in the server I think)

If not, you can always add them into the Earth wiki:

https://earth.bsc.es/wiki/doku.php?id=tools:autosubmit

Media Manager...

**Comment by LuiggiTenorioK** [comment]
In GitLab by @kinow on Oct 26, 2022, 09:51

Assuming the images are not really big (>4MB or so I guess), we can upload them directly in the git repository, in the `html_static_path` or in any other location of the source directory.

The `html_static_path` is normally `_static` in Sphinx projects, e.g. in CWL:https://github.com/common-workflow-language/user_guide/blob/0b55dc767ecd8b288d3ffa3e6f24afbf59864f5f/src/conf.py#L158  and in Cylc: https://github.com/cylc/cylc-doc/blob/9304c0837d75bfcfcd6e0b09bc0a52f12c9f0920/src/conf.py#L183

In the CWL User Guide the logo is rendered from a subfolder of `html_static_path`: https://github.com/common-workflow-language/user_guide/blame/0b55dc767ecd8b288d3ffa3e6f24afbf59864f5f/src/introduction/basic-concepts.md#L15

But images can reside elsewhere too. In the case of Cylc, for instance, the images are stored in subdirectories of the sections. Their `tutorial` folder has an `img` folder for images: https://github.com/cylc/cylc-doc/tree/master/src/tutorial/img

Then a directive is accessing that location to load images, e.g. https://github.com/cylc/cylc-doc/blob/master/src/tutorial/scheduling/integer-cycling.rst#recurrence-sections

That's one way of doing it... we can store images anywhere with the source code. Some images might go to static (e.g. logo, favicons), while others may be stored near the sections... or they could be stored elsewhere too if we prefer I think.

IMHO, the disadvantage of using an external service is that users wouldn't be able to build and run the documentation offline (e.g. if someone cloned the project in some area of the HPC, or they are in a flight working offline, etc.).

**Comment by LuiggiTenorioK** [comment]
In GitLab by @kinow on Oct 26, 2022, 09:55

I mentioned dynamic images… here's an example, also from the CWL User Guide: https://github.com/common-workflow-language/user_guide/blame/0b55dc767ecd8b288d3ffa3e6f24afbf59864f5f/src/introduction/basic-concepts.md#L51-L103

This is used to generate [an SVG graph](http://www.commonwl.org/user_guide/introduction/basic-concepts.html#implementations) during build time in ReadTheDocs that is added to the documentation:

![image](https://earth.bsc.es/gitlab/es/autosubmit-gui/uploads/8d570414fe570b80dd7b4cfa4074f538/image.png)

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Oct 26, 2022, 11:14

Yeah, it worked fine if we uploaded to the repo all images ([Autosubmit's repo](https://autosubmit.readthedocs.io/en/latest/usage/run_modes/wrappers.html#hybrid-wrapper) is like this). But as far as I know (not much) it is not advised to upload to git images. But if both you and @mcastril are in favor of uploading images to the repo, I'll do it now and solve this issue.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 26, 2022, 16:28

Hi @manuel-g-castro , maybe you can fork the main branch and add the images there temporarily and use that new branch to compile and generate the documentation. then we we finally decide the best approach we take the right approach and delete this branch and use the master again

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Oct 27, 2022, 09:17

Done, @ jberlin. Created new branch [with-images](https://earth.bsc.es/gitlab/es/autosubmit-gui-docs/-/tree/with-images) at the [docs repo](https://earth.bsc.es/gitlab/es/autosubmit-gui-docs) and activated the branch at readthedocs. [You can check it out](https://autosubmit-gui.readthedocs.io/en/with-images/statistics/index.html).

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Oct 27, 2022, 10:11

Thanks @manuel-g-castro

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Nov 2, 2022, 07:58

Hi @ jberlin , 

Regarding the deployment situation we could go the easy way (once I deploy the latest changes regarding Issue #47 ) and just push to master the **current active versions of the API and the GUI that are on the Rocky env dev** and delete the other branches such as the one I created to work while the API user log in was not available.

What do you think?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 2, 2022, 10:05

Hi @ggcr , is not bad idea but we will run into additional merges and some parts of the code may also be dirty (debugging alerts messages, commented/dead code, etc) and merges in console are usually error-prone . 

Some of the changes need code-review before actually putting them to master so please after #47 is completed submit the branch for CR and assign it to me, thanks!. After we merge these changes I will do a clean checkout of the master branch and deploy both the API and the GUI in dev.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Nov 3, 2022, 08:42

Sorry I thought I had to assign you as reviewer only and not also as assignee. It's done now.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Nov 3, 2022, 16:32

Merged with-images branch to the master.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 3, 2022, 18:59

Yes, I think it's better to be conservative and do the usual merges and reviews, although it's more work, it's safer.

After all the merges I agree that all the other non-merged and legacy branches should be removed if they are not aiming to be merged in master.

If you find the develop/commit/deploy process slow, we can talk about alternatives.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 3, 2022, 19:02

Perfect, I see that [latest](https://autosubmit-gui.readthedocs.io/en/latest/graph/index.html) contains the images. I guess that [with-images](hedocs.io/en/with-images/graph/index.html) has to be removed from RTD.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Nov 5, 2022, 13:24

I thought that the branch would be deleted. I disabled the branch.

![image](https://earth.bsc.es/gitlab/es/autosubmit-gui/uploads/78faf82eeb1f8737871c9dfee1419032/image.png)

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Nov 9, 2022, 07:53

It's fine for me, I'll try to be more declarative and to make more independent commits (one for each functionality for example) so that the review process gets easier :smile:

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 9, 2022, 17:18

marked the checklist item **Generate the API new version and upload to Pypi.** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 9, 2022, 17:19

marked the checklist item **Update GUI and deploy API in development environment for the final validation** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 9, 2022, 17:25

I generated TAGs in both projects to mark the end of this release:

- AS_GUI_SP1
- AS_API_SP1

I deployed both TAGs in the development environment, until we close all the issues in the milestone, we should not change anything. we can validate the issues tomorrow or in the upcoming days

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 9, 2022, 18:17

I also updated deployment wiki for AS API, to make it more clear

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 9, 2022, 18:37

Yes, this is important. They will be squashed anyway at the time of merging them.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 9, 2022, 18:55

Good idea to create those tags. 

I think it would be useful to add the list of new features in the tag description. It will be very helpful in the future for many reasons: to justify your work; to do a recap in dissemination sessions, workshops etc; to have a track of the work done at the end of the year...

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 10, 2022, 13:23

mentioned in commit 5bd89f74b25dacb87d00f9518a0a8f90bac51459

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 16, 2022, 12:59

I reviewed the wiki and I think it lack's info about the development and production environments.

In the [GUI wiki](https://earth.bsc.es/gitlab/es/autosubmitreact/-/wikis/Deployment-and-Update) there's no mention about it.

In the [API wiki](https://earth.bsc.es/gitlab/es/autosubmit_api/-/wikis/Deployment-and-Update) there is some mention to `rocky` but then I am not sure if the paths refer to the development or production environment.

To add this information, I think the best is that you create a section in the `API wiki` explaining the production and development environment (hosts, users, directories, etc).

Then, in the deployment and update sections' for both wikis, explain the depoyment-testing-deployment process involving the two machines, referring to the new wiki section and ensuring that the paths are consistent. If the paths are already explained and abstracted in the new wiki section, they can be just referred with their abstract name or variable.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 16, 2022, 13:33

Hi @mcastril , yes that is true, we can enrich the current documentation with more details of each environment, could be very useful for deploy

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 16, 2022, 17:13

Hi @mcastril, I added in both wikis a new section called `environments` where there is a table with the most important things to know and some other useful information

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 18, 2022, 12:53

marked the checklist item **Final validation of the release, Close all issues in the milestone, both in the GUI and API projects.** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 18, 2022, 12:53

marked the checklist item **Final validation of the release, Close all issues in the milestone, both in the GUI and API projects.** as incomplete

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 18, 2022, 12:57

marked the checklist item **Final validation of the release, Close all issues in the milestone, both in the GUI and API projects.** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 18, 2022, 13:03

All involved issues were validated in the development VM, so we can start thinking on deploy this new release towards the end of the month (when I am present), we should:

- Send email to the user that X day we will be deploying a new version and some short disruption can occur
- Send email to communicate the new features of AS GUI & improvements/fixes made 

I was thinking to deploy this new version on **Monday 28 after 18 hours**, so we dont cause much hassle to our users, what do you think @mcastril ?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 18, 2022, 16:21

It's a good idea @ jberlin. I think that all the info can be in the same e-mail.

You and @ggcr can start writing an email here that we can edit together.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 18, 2022, 17:55

I recreated the TAGs in both projects: v1.5.0 , I think we should maintain original versioning in both projects so we know which version of API goes with which version of the GUI

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Nov 18, 2022, 18:25

Sounds good! @ jberlin 

I don't know neither the tone and the type of writing of this type of mails.

Should it be like a Changelog?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 28, 2022, 12:29

Hi @mcastril , @ggcr , something like this I had in mind for the announcement of the new release and to warn the user of some potential disruptions when we deploy the new version:

```
Dear Autosubmit users.

Today we are pleased to announce the deployment of a new release of Autosubmit GUI. The following new features have been added:

- Treeview UI/UX improved by adding new folders for the wrappers, and inside these, we added a sorting to display the proper execution sequence, collapse all/Expand all buttons were added
- New tab icon was added to replace generic React.js icon that was previously being displayed
- An early initial load of the Active experiments at the homepage has been implemented without requiring the user to click on the button each time.
- UI/UX of the graph view improved: Tasks without dependencies are displayed correctly in the Graph View in a single row now. now also is possible to select a group of nodes by keeping pressed the right button of the mouse and hover the rectangle over these.

Apart of these new features, several performance improvements and bug fixes were carried out, for the full list of improvements, you can check the [news section](https://earth.bsc.es/autosubmitapp/news) of the Autosubmit GUI.

The deployment will be carried out on Thursday 01 of December, so expect some minor disruptions around 17/18 hours.
```

Please feel free to provide feedback, I would like to send it today if possible

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Nov 28, 2022, 12:36

The UI/UX graph section is missing the right-click square node selection.

For the rest, I honestly think it's very well redacted

**Comment by LuiggiTenorioK** [comment]
In GitLab by @manuel-g-castro on Nov 28, 2022, 12:46

I would reorder and remove some topics to bring more attention to the new stuff. Something like

```
Dear Autosubmit users:

Today we are pleased to announce the deployment of a new release of Autosubmit GUI. The following new features have been added:

- Treeview UI/UX 
  - Added new folders for the wrappers
  - Inside folder sorting to display the proper execution sequence
  - Collapse/Expand buttons
- Graph view
  - Tasks without dependencies are displayed in a single row 
- A new initial load of the Active experiments at the homepage
- New tab icon (REALLY IMPORTANT)

Finally, performance improvements and bug fixes.

The deployment will be carried out on Thursday 01 of December from 17h to 18h, so expect some minor disruptions.
```

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 28, 2022, 14:49

Hi @manuel-g-castro , good idea, indeed the completed set of changes is in news section in the app, we can add also that more details can be readed in the news section, and I also will add what you mentioned @ggcr , thanks for the help

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 28, 2022, 17:35

mentioned in commit 480d786a9ee615238f812cd9ca65e5d041a61f87

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 28, 2022, 18:10

thanks all for the feedback, the final msg is as follows:

```
Dear Autosubmit users.

Today we are pleased to announce the deployment of a new release of Autosubmit GUI. The following new features have been added:

- Treeview UI/UX improved by adding new folders for the wrappers, and inside these, we added sorting to display the proper execution sequence, collapse all/Expand all buttons were added
- New tab icon was added to replace the generic React.js icon that was previously being displayed
- An early initial load of the Active experiments at the homepage has been implemented without requiring the user to click on the button each time.
- UI/UX of the graph view improved: Tasks without dependencies are displayed correctly in the Graph View in a single row now. now also is possible to select a group of nodes by keeping pressed the right button of the mouse and hovering the rectangle over these.

In addition to these new features, several performance improvements and bug fixes were also carried out, for the full list of improvements, you can check the [news section](https://bscesautosubmitdev01.bsc.es/autosubmitapp/news) of the Autosubmit GUI (DEV).

The deployment will be carried out on Thursday 01 of December, so expect some minor disruptions around 17/18 hours.

Autosubmit Team
```

I added the link to the news page of the VM for more detailed info

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 28, 2022, 18:15

mentioned in commit 3a16a923eecd2efd596f0cb8df8a975f1f98a603

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 29, 2022, 17:34

I prepared the build folders and versioning of the upcoming release in both GUI and API projects, the internal version will be v1.7.0.

https://earth.bsc.es/gitlab/es/autosubmit_api/-/tags/v1.7.0

https://earth.bsc.es/gitlab/es/autosubmitreact/-/tags/v1.7.0

now I understand the purpose of the build folders, there is no npm in the web server so it cannot be built over there, it must be built beforehand. then I updated these folders, version files, and production settings and regenerated the tags.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 30, 2022, 09:31

marked the checklist item **Prepare a message to send to the BSC-ES mailing list, with a brief summary of the new features (and also reminding abut the best advantages of using the GUI)** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 30, 2022, 09:35

if there are no objections or showstoppers, I will send the email we wrote and proceed with the deployment tomorrow Thursday around lunchtime (optimal time since CP/AC people go to lunch usually).

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 30, 2022, 17:54

I support @manuel-g-castro idea of splitting the changes into short sentences. Most of the users will look over the message in 10 seconds, and short sentences are easier to catch and more attractive to procrastinators.

Besides the wrapper grouping, the collapse functionality also brought the summary tooltips to the startdate level too. Not sure if this is advertised.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 30, 2022, 17:57

I think Tuesday is a better day since Thursdays are busy with meetings, and we have less time to react in case of issues, but as next Tuesday is a bank holiday, I think it's ok.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Nov 30, 2022, 17:59

Hi @mcastril, yes exactly, I would prefer Tuesday but next week, Tuesday is a public holiday, I think if we do it during lunchtime it should be OK.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Nov 30, 2022, 18:00

In any case, we should have some deployment scripts to backup the production directory and upload the build to the same directory as we discussed. If the building process is remote, the script could be run somewhere else and do the backup / copy the new build over SSH.

We would need a similar script to roll back (delete or backup the production folder and substitute it with the last backup) in case of an emergency.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 09:43

Hi @mcastril , yes true, I thought it was there the caption in the top level of the start dates but it isn't I will add it and make it shorter

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 09:47

I think the GUI is easy, for the API we can uninstall the latest version and install again the previous one with PIP:

```pip install <AS_API_PACKAGE>==<PREVIOUS_VERSION>```

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 13:35

marked the checklist item **if validation in VM was OK, deploy in production, before that send email to distributions lists to warn the users of the update/possible disruptions** as completed

**Comment by LuiggiTenorioK** [comment]
In GitLab by @eferre1 on Dec 1, 2022, 14:27

Hi @ jberlin, I've tried the new expand/collapse all buttons, but the screen goes white and doesn't show anything on my workstation (using firefox), but then on my laptop (using chrome) it works fine without error. Is it something related to cleaning the cache or how should I proceed? (Also to take it into account for other users that might have this issue)

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 14:29

hi @eferre1 , which experiment have you tried ? so I can check it out

**Comment by LuiggiTenorioK** [comment]
In GitLab by @eferre1 on Dec 1, 2022, 14:31

`a3pg`

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 14:33

hi @eferre1 , yes it happens to all experiments as far as I checked , I will create an issue since this only happens in Firefox, thanks for checking

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Dec 1, 2022, 14:42

In my case, there are different experiments that I don't see the buttons, out of the one reported in https://earth.bsc.es/gitlab/es/autosubmit_api/-/issues/8#note_189881

PS: It's fixed after reloading (ctrl+R) but after switching to another experiment I don't see the buttons again. It disappeared after I reloaded the page in 3 or 4 experiments. We can add some notes about caching in the next releases.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @eferre1 on Dec 1, 2022, 14:45

Ok, I tried cleaning the cache since some time that was the problem after a GUI update, but it doesn't seem to have had an effect this time.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Dec 1, 2022, 14:53

The buttons are working fine in Firefox 107.0 (64-bit). Which version are you using @eferre1 @ jberlin ?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 14:54

I am using FF 91.0.1esr (64-bit)

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Dec 1, 2022, 14:54

I will try to install the latest one, although I am not sure if we have rights for that in the BSC WS

**Comment by LuiggiTenorioK** [comment]
In GitLab by @eferre1 on Dec 1, 2022, 14:56

Same, 91.0.1esr version, using it from the WS as well.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @mcastril on Dec 1, 2022, 15:01

I am using my laptop. You cannot change the WS version through package system, only install a new one in your /home, for example (manually).

I don't think it's worth it. Let's focus on fixing the problem in the WS version. Can you send a message to the **autosubmit list** saying that the Expand/Collapse buttons are broken in the Firefox WS version (but working fine in Chrome or recent Firefox versions) until further notice?

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ggcr on Dec 1, 2022, 16:33

I haven't checked yet but I'm pretty sure this had to do with the fact that I expand the nodes with a recursive function and some browsers don't like that. I'll try the iterative one.

**Comment by LuiggiTenorioK** [comment]
In GitLab by @ jberlin on Jan 23, 2023, 16:39

marked the checklist item **Validation in production, checking of the solved the issues & new features.** as completed
